kms-message/src/kms_message_private.h

```diff
@@ -46,6 +46,20 @@ struct _kms_request_t {
    bool auto_content_length;
    _kms_crypto_t crypto;
    kms_request_provider_t provider;
+
+   /* TODO (MONGOCRYPT-???): make a union for each KMS provider type.
```

Comment:
I think a union is appropriate for this. I did not want to modify existing fields to keep the diff small. If there is consensus, I will create a MONGOCRYPT ticket and update the comment.

Reply:
Agreed, I think a union is a good idea.

Reply:
+1

Reply:
Filed MONGOCRYPT-342